src/strategy/periodic.rs

```diff
@@ -41,15 +58,57 @@ impl StrategyPeriodic {
             n => log::trace!("periodic updates, weekly calendar length: {} minutes", n),
         };
 
-        let strategy = Self { schedule: calendar };
+        let strategy = Self {
+            schedule: calendar,
+            time_zone: tz,
+        };
         Ok(strategy)
     }
 
+    /// Get the time zone from `periodic` strategy config.
+    fn get_time_zone(cfg: &inputs::PeriodicInput) -> Fallible<Tz> {
+        let tz;
+        if &cfg.time_zone == "localtime" {
+            // Use `read_link` instead of `exists()` because we only want to check for
+            // the existence of the `/etc/localtime` symlink, not whether it points to
+            // a valid location. (`read_link` returns an error if symlink doesn't exist)
+            if let Err(_) = read_link(Path::new("/etc/localtime")) {
+                tz = Tz::named("UTC");
+            } else {
+                tz = Tz::local();
+            }
+        } else {
+            tz = Tz::named(&cfg.time_zone)
+        }
+        tz.map_err(|err| err.into())
+    }
+
     /// Return the measured length of the schedule, in minutes.
     pub(crate) fn schedule_length_minutes(&self) -> u64 {
         self.schedule.length_minutes()
     }
 
+    /// Return the weekday and time of the next window, in human terms.
+    pub(crate) fn human_next_window(&self) -> String {
+        let naive_utc_dt = Utc::now().naive_utc();
+        let dt = (&self.time_zone).from_utc_datetime(&naive_utc_dt);
+        let next_window_minute_in_week = self.schedule.next_window_minute_in_week(&dt);
+
+        match next_window_minute_in_week {
+            Some(minute_in_week) => {
+                let (weekday, hour, minute) = utils::weekly_minute_as_weekday_time(minute_in_week);
+                // `tmp_dt` is created just to use the `%Z` specifier of its `format()` method.
+                let tmp_dt = (&self.time_zone).from_utc_datetime(&Utc::now().naive_utc());
+                let tz_abbrev = tmp_dt.format("%Z").to_string();
```

Comment:
This is a hacky way to get an abbreviation or otherwise some sort of name of the configured time zone. Ideally, `tzfile::Tz` should implement `Display`.

Reply:
This is possibly something to bring to upstream, however it may be hard to sell as timezones don't have a canonical/preferred name (see for example `UTC` vs `Etc/UTC`).

Reply:
Good point. Perhaps don't implement `Display`, but maybe have a `name()` method that returns the last bit of the path of the `tzfile(5)` that the `tzfile::Tz` struct reads its data from, e.g. two different `Tz` instances may return `America/Toronto` or `Canada/Eastern` even though the data should be the same.